bowtie/_core.py

```diff
@@ -538,6 +543,43 @@ def supports(self, *dialects: Dialect) -> bool:
         """
         return self.info.dialects.issuperset(dialects)
 
+    async def get_versions(self):
+        from github3 import GitHub  # type: ignore[reportMissingTypeStubs]
+        from github3.exceptions import (  # type: ignore[reportMissingTypeStubs]
+            GitHubError,
+        )
+        from github3.models import (  # type: ignore[reportMissingTypeStubs]
+            GitHubCore,
+        )
+
+        url = CONTAINER_PACKAGES_API / self.id / "versions"
+        versions: Set[str] = set()
+
+        pages: list[GitHubCore] = []
+        try:
+            gh = GitHub(token=os.environ.get("GITHUB_TOKEN", ""))
+            pages = gh._iter(count=-1, url=str(url), cls=GitHubCore)  # type: ignore[reportPrivateUsage]
+        except GitHubError:
+            pass
+
+        for page in pages:
+            try:
+                tags = cast(
+                    Iterable[str],
+                    page.as_dict()["metadata"]["container"]["tags"],
+                )
+            except KeyError:
+                continue
+            else:
+                versions.update([tag for tag in tags if "." in tag])
+
+        if not versions and self.info.version:
+            versions.add(self.info.version)
+
+        return frozenset(
+            sorted(versions, key=sortable_version_key, reverse=True),
+        )
```

Comment:
Also @Julian I am facing an issue over here in these lines above, when we try to do `frozenset(sorted(...))`, th e sorted order is not maintained after converting it to a `frozenset`. Is there a way of fixing this?

Reply:
`frozenset` indeed isn't ordered. You have to pick one -- either return a set and let the person using this sort, or just return the iterable.

Reply:
I tried a bit of a hack by following a discussion that I got on Stack Overflow and the below way seemed to work and also maintained the sorted order while returning a `frozenset`

```python
return frozenset(
    zip(
        *zip(sorted(versions, key=sortable_version_key, reverse=True)),
    ),
)
```

just that now I have to do 

```python
metadata.append(("versions", next(iter(await each.get_versions()))))
```

Reply:
So now `get_versions` is returning the below

```python
async def get_versions(self) -> frozenset[zip[tuple[str, ...]]]
```

Reply:
I don't understand why you need/want to do this though. Why do you need both here?

Reply:
That kind of StackOverflow answer is absolutely horrid. If you wanted both, use the dependency you found.

Reply:
> frozenset indeed isn't ordered. You have to pick one -- either return a set and let the person using this sort, or just return the iterable.

Okay I guess I will go with just returning a set then. Using `frozenset` with `SortedSet` also doesn't seem to be maintaing the order so no point in adding that dependency either.